browser/ntp_background/view_counter_service_factory.cc

```diff
@@ -79,6 +80,9 @@ ViewCounterServiceFactory::BuildServiceInstanceForBrowserContext(
         browser_context, std::make_unique<NTPBackgroundImagesSource>(service));
     content::URLDataSource::Add(
         browser_context, std::make_unique<NTPSponsoredImagesSource>(service));
+    content::URLDataSource::Add(
```

Comment:
I think my other comment got cleared with the weakptr change, but why are we adding these in ViewCounterServiceFactory? Shouldn't these be added in the webui? This is definitely not the right place for this

Reply:
This is based on legacy code as you can see just above our `Add`, ok to raise a followup? Thanks

Reply:
follow-up is fine for the others, but we shouldn't add this new one here

Reply:
Raised https://github.com/brave/brave-browser/issues/43828